extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerConfig.java

```diff
@@ -19,8 +20,52 @@
     /** Build / static runtime config for span exporters */
     public SpanExporterConfig exporter;
 
+    /**
+     * Tracing ID generator.
+     * <p>
+     * Valid value is {@code xray}.
+     * <p>
+     * If not set defaults to the default OpenTelemetry ID Generator.
+     */
+    @ConfigItem
+    public Optional<IdGenerator> idGenerator;
+
+    /**
+     * A comma separated list of name=value resource attributes that
+     * represents the entity producing telemetry
+     * (eg. {@code service.name=authservice}).
+     */
+    @ConfigItem
+    public Optional<List<String>> resourceAttributes;
```

Comment:
I'm uncertain of the value in enabling any key/value pair to be set on the `Resource`.

Reply:
Here we have a use case to override the application name for instance, and this can be picked up by env variables. We use AWS XRay as the tracing backend, and we have multiple deployments of the same service in the same AWS region (XRay is shared in the same region) so we have a pattern in the tracing service name that also takes that into account, and because of that we need to set values by deployment. When adding this I also made it this way to be following the approach that OTEL sdk was using

Reply:
Am I correct in saying these configuration items should be runtime settable?

At the moment they're build time config only, so I think that probably should change

Reply:
Yup =]

Reply:
I removed that right now, since we can set that by using a resource provider SPI, or if we move to beans, we can create a bean for doing the same thing